kore/src/Kore/Repl/Data.hs

```diff
@@ -158,11 +161,11 @@ newtype ReplOutput = ReplOutput {unReplOutput :: [ReplOut]}
     deriving newtype (Semigroup, Monoid)
 
 -- | Newtypes for printing functions called by Kore.Repl.Interpreter.replInterpreter0
-newtype PrintAuxOutput = PrintAuxOutput
-    {unPrintAuxOutput :: String -> IO ()}
+data PrintAuxOutput = PrintAuxOutput
+    {unPrintAuxOutput :: forall m. MonadIO m => String -> InputT m ()}
```

Comment:
Do we need this generality or could this just be `String -> InputT IO ()`? Same for `PrintKoreOutput`. If this wasn't existential before, I don't think it would need to be now.

Reply:
Unfortunately we do need the existentials. Notice the return type of `replInterpreter0` https://github.com/runtimeverification/haskell-backend/blob/ee048b50eec9f1eb15ba7a9d8fc51d9fdd4a3b8e/kore/src/Kore/Repl/Interpreter.hs#L280

when we traverse the output within the function body: https://github.com/runtimeverification/haskell-backend/blob/ee048b50eec9f1eb15ba7a9d8fc51d9fdd4a3b8e/kore/src/Kore/Repl/Interpreter.hs#L330-L335

I see no (nice) way to lift `InputT IO ()` to `InputT (ReaderT (Config m) (StateT ReplState m)) ()`, since the haskeline library doesn't export the `InputT` constructor. 

I was originally going to do what you suggest and add a function 

```haskell
liftInputT :: MonadIO m => InputT IO a -> InputT m a
```
to massage the types into place, but alas...